firefox-ios/Client/Utils/GleanOhttpUploader.swift

```diff
@@ -0,0 +1,86 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+import Glean
+import MozillaAppServices
+import Shared
+
+protocol ASOhttpManager {
+    func data(for request: URLRequest) async throws -> (Data, HTTPURLResponse)
+}
+
+extension OhttpManager: ASOhttpManager {}
+
+/// An enumeration representing different environments for the OHTTP client.
+/// This is an enum in case we want to add other environment other than production.
+enum OhttpEnvironment {
+    case prod
+
+    /// Returns the configuration URL based on the selected environment.
+    var config: URL? {
+        switch self {
+        case .prod:
+            return URL(string: "https://prod.ohttp-gateway.prod.webservices.mozgcp.net/ohttp-configs")
+        }
+    }
+
+    /// Returns the relay URL based on the selected environment.
+    var relay: URL? {
+        switch self {
+        case .prod:
+            return URL(string: "https://mozilla-ohttp.fastly-edge.com/")
+        }
+    }
+}
+
+struct GleanOhttpUploader {
+    private let connectionTimeout = TimeInterval(10)
+    private let manager: ASOhttpManager
+    private let logger: Logger
+
+    init(manager: ASOhttpManager,
+         logger: Logger = DefaultLogger.shared) {
+        self.manager = manager
+        self.logger = logger
+    }
+
+    /// Build the request and create upload operation using the `OhttpManager`
+    func uploadOhttpRequest(request: GleanPingUploadRequest,
+                            callback: @escaping (UploadResult) -> Void) {
+        guard let url = URL(string: request.url)
+        else {
+            logger.log("Rejected ohttp ping upload due environment variables", level: .info, category: .telemetry)
+            callback(.unrecoverableFailure(unused: 0))
+            return
+        }
+
+        var body = Data(capacity: request.data.count)
+        body.append(contentsOf: request.data)
+
+        var oHttpRequest = URLRequest(url: url,
```

Comment:
Can this use your same `buildRequest` as above? The interior request should built in the same way. While the main difference is if we use `manager.data` vs `session.uploadTask`